### PR TITLE
send a broadcast action to other apps

### PR DESCRIPTION
**Pull request by vitas**
Hi,
Just added a ability to broadcast intent by string action. Useful if you want exec some app or their intent, for example i#m sending an action to the app what sends some notifications.
We could consider to add support of some apps like Tasker

### CODE REVIEW

**Comment by sylvek** [comment]
:+1:

File: app/src/main/java/net/sylvek/itracing2/receivers/CustomAction.java [hunk 1]
```diff
@@ -31,6 +31,10 @@ public void onReceive(Context context, Intent intent)
         if (action.startsWith("http://")) {
             new CallUrl<HttpsURLConnection>(action).start();
         }
+
+        if (!action.isEmpty()) {
```
Comment (sylvek) [inline app/src/main/java/net/sylvek/itracing2/receivers/CustomAction.java]: you could use a pattern like "intent://..." and adapt your conditional loop.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -45,13 +45,13 @@
     <string name="double_button_delay_summary">delay between two clicks over button</string>
     <string name="default_double_button_delay">300</string>
     <string name="custom_action">custom action</string>
-    <string name="custom_action_summary">http(s)://..</string>
+    <string name="custom_action_summary">http(s)://.. or intent action string</string>
```
Comment (sylvek) [inline app/src/main/res/values/strings.xml]: or intent://.. 

for example.

**Comment by sylvek** [comment]
that's a really a good thing :)
do you have any use cases ?

**Comment by vitas** [comment]
well, just use it for one custom app, but any serious app has set of intent actions what are public available. You can add "tasker" support. take a look at this poweful app

File: app/src/main/java/net/sylvek/itracing2/receivers/CustomAction.java (hunk 1 above)
Comment (vitas) [inline app/src/main/java/net/sylvek/itracing2/receivers/CustomAction.java]: not sure you need that complexity now, some actions string could be long  and user will have to add this prefix..

**Pull request merged**